library/pkcs7.c

```diff
@@ -312,6 +313,7 @@ static int pkcs7_get_signer_info(unsigned char **p, unsigned char *end,
         goto out;
     }
 
+    end_issuer_and_sn = *p + len;
     /* Parsing IssuerAndSerialNumber */
     signer->issuer_raw.p = *p;
 
```

Comment:
`end_signer` should be replaced by `end_issuer_and_sn` in various places.  This is harmless inasmuch as any mismatches will be caught by the `*p != end_issuer_and_sn` check.